Xamarin.Forms.Core/Routing.cs

```diff
@@ -84,6 +84,21 @@ internal static string GetRoutePathIfNotImplicit(Element obj)
 			return $"{source}/";
 		}
 
+		internal static Uri RemoveImplicit(Uri uri)
+		{
+			if (!uri.IsAbsoluteUri)
+				return uri;
+
+			string[] parts = uri.ToString().TrimEnd('/').Split('/');
```

Comment:
Original string is a good suggestion but I don't want to remove the empty entries here. Otherwise if you have "app://route" it will turn into "app:route"

I don't think we need to have the backslashes do we? If it's an absolute uri at this point it'll be forward slashes right?

Reply:
Absolute path can contain a slash and backslash same time and it will be correct. Users can enter as they like. We have a 100% certainty that the input Uri will only use slashes?

Reply:
alright I added code to sanitize absolute uris

https://github.com/xamarin/Xamarin.Forms/pull/5954/commits/1da11a1fc406c436e6e1073bea2867cf82afab51#diff-e3b95c85373e6c0fb300485802be7ab6R18

I just really don't want to have "/", "\\" everywhere inside internal code :-)